proposals/p2015.md

```diff
@@ -0,0 +1,237 @@
+`Sum` as a 32-bit two's complement signed integer.
+
+### Syntax examples and considerations
+
+These examples are not meant to be formal mappings. They are only used to
+illustrate to the reader how the numeric type syntax may connect to the
+underlying language semantics in future numeric type additions.
+
+#### Syntax example use in common fixed-size numeric types
+
+[As proposed by @chandlerc](https://github.com/carbon-language/carbon-lang/issues/543#issuecomment-845620894),
+our heuristic for type qualification here will be "Any truly pervasive
+non-aggregate/composite/compound type that is also fundamentally
+size-parameterized."
+
+These fixed-width sizes can naturally cover the common register widths of 8-,
+16-, 32-, 64-, and 128-bits, as seen in other languages.
+
+Which expands to:
+
+-   `i8`, `i16`, `i32`, `i64`, and `i128` keyword aliases for
+    `Carbon.Int(N={8,16,32,64,128})`
+-   `u8`, `u16`, `u32`, `u64`, and `u128` keyword aliases for
+    `Carbon.UInt(N={8,16,32,64,128})`
+-   `f16`, `f32`, and `f64` keyword aliases for `Carbon.Float(N={16,32,64})`
+
+Additionally, other type sizes may be available, depending on the platform (for
+example, `x86`), such as `f80`, `f128`, or `f256`.
+
+#### Syntax example use in arbitrary fixed-size numeric types
+
+In the more general case, this syntax can express N-width types.
+
+-   `iN` for `Carbon.Int(N)`, An N-width two's complement signed integer type
+-   `uN` for `Carbon.UInt(N)`, An N-width unsigned integer type
+-   `fN` for `Carbon.Float(N)`, An N-width IEEE 754 floating-point number type
+
+## Rationale
+
+Following Carbon's goal to facilitate
+["Code that is easy to read, understand, and write"](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#code-that-is-easy-to-read-understand-and-write),
+an explicit goal is to provide excellent ergonomics.
+
+Highlighting relevant aspects of this from the project goals:
+
+-   _Carbon should not use symbols that are difficult to type, see, or
+    differentiate from similar symbols in commonly used contexts._
+-   _Syntax should be easily parsed and scanned by any human in any development
+    environment, not just a machine or a human aided by semantic hints from an
+    IDE._
+-   _Explicitness must be balanced against conciseness, as verbosity and
+    ceremony add cognitive overhead for the reader, while explicitness reduces
+    the amount of outside context the reader must have or assume._
+
+The type system syntax must also complement Carbon's target for
+["Performance-critical software"](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#performance-critical-software)
+
+Specifically, there should be "No need for a lower level language."
+
+-   _Developers should not need to leave the rules and structure of Carbon,
+    whether to gain control over performance problems or to gain access to
+    hardware facilities._
+
+## Alternatives considered
```

Comment:
Add an alternative about different sets of bit sizes that could be supported (all bit sizes, just common powers of 2).